src/aws-cpp-sdk-core/source/auth/AWSCredentialsProviderChain.cpp

```diff
@@ -34,6 +34,9 @@ AWSCredentials AWSCredentialsProviderChain::GetAWSCredentials()
         AWSCredentials credentials = credentialsProvider->GetAWSCredentials();
         if (!credentials.GetAWSAccessKeyId().empty() && !credentials.GetAWSSecretKey().empty())
         {
+            // TODO: issue of only chain, not overidden
```

Comment:
whats this todo on about? we should also override `GetAWSCredentials(Aws::AmazonWebServiceRequest& request)` on the credentials provider chain and pipe in the request from the client otherwise I dont think this will work as intended